### PR TITLE
feat: bring back browser extension

### PR DESCRIPTION
**Pull request by svrnm**
## Which problem is this PR solving?

This PR brings back the browser extension by fixing the build errors & updating dependencies and some, see #1101, #1110 

## Short description of the changes

- Move extension back to packages
- Update dependencies until `npm run compile` works again
- Update some of the dependencies and words (e.g. move from "Collector Trace" to "OTLP HTTP" exporter, etc.)
- Add some clean ups
- Add myself as component owner
- Add a `server` folder that contains instructions on how to run a collector behind a nginx for proper CORS headers.

### CODE REVIEW

**Comment by svrnm** [comment]
not sure how to fix the issues with node 8 & node 10. Best option would be to exclude them, since the final runtime is the browser and the testing doesn't need to run on all node versions.

File: packages/opentelemetry-browser-extension-autoinjection/package.json
Comment (Flarna) [inline packages/opentelemetry-browser-extension-autoinjection/package.json:3]: I don't think it's needed to change the version here as this should be done by the release automation.

It might be needed to re-add this module in .release-please-manifest.json to let the release automation know about the last release. But I'm not sure here.

@dyladan  I think you know most about release-please. Do you know if adding to .release-please-manifest.json is required?

Comment (dyladan) [reply packages/opentelemetry-browser-extension-autoinjection/package.json:3]: yes it is required. it is also required to add to `.release-please-manifest.json`

Comment (svrnm) [reply packages/opentelemetry-browser-extension-autoinjection/package.json]: Reset the version to 0.29 & added the package to .release-please-manifest.json

**Review by Flarna — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Comment by dyladan** [comment]
Failure appears to be a dependency conflict due to webpack versioning differences with other packages in the repo

**Comment by svrnm** [comment]
> Failure appears to be a dependency conflict due to webpack versioning differences with other packages in the repo

looks like I still have to learn about lerna and stuff... let me see if I can get this working somehow, because updating to a more recent version of webpack did the trick to get it compiled again, and downgrading sends me into a rabbit hole of dependency issues.

**Comment by blumamir** [comment]
@MSNev aligned the webpack versions in #1264 
Maybe rebasing main will fix the CI?

**Comment by blumamir** [comment]
Also related - renovate's #1258 and #1123 

Might be a good chance to upgrade these as well while we are at it, if you can

File: packages/opentelemetry-browser-extension-autoinjection/package.json
Comment (blumamir) [inline packages/opentelemetry-browser-extension-autoinjection/package.json:30]: Why upgrade the "@opentelemetry/api" minimum supported minor versions?

If the user is using an older version of the API, will it fail to work?

Comment (svrnm) [reply packages/opentelemetry-browser-extension-autoinjection/package.json:30]: Not sure why I changed it, it _should_ work with 1.0 as well

**Comment by svrnm** [comment]
> @MSNev aligned the webpack versions in #1264 Maybe rebasing main will fix the CI?
> [...]
> Also related - renovate's https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1258 and https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1123
> Might be a good chance to upgrade these as well while we are at it, if you can

Unfortunately no, while fixing the initial issue with the browser extension I upgraded to webpack 5, but learna's hosting brings it down to a 4.x version and then things break

**Comment by svrnm** [comment]
As discussed with @blumamir I am still keen to bring the extension back somehow, but I would need some guidance how to go about it.

**Comment by blumamir** [comment]
We talked about it in the last SIG and there is support for upgrading to webpack 5.
There have been previous attempts in the past but they were unsuccessful because of dependencies incompatible with node8. Now that the repo supports node >=14 it should be ok.

**Comment by svrnm** [comment]
> We talked about it in the last SIG and there is support for upgrading to webpack 5. There have been previous attempts in the past but they were unsuccessful because of dependencies incompatible with node8. Now that the repo supports node >=14 it should be ok.

That's awesome, happy to hear that!

**Comment by blumamir** [comment]
> > We talked about it in the last SIG and there is support for upgrading to webpack 5. There have been previous attempts in the past but they were unsuccessful because of dependencies incompatible with node8. Now that the repo supports node >=14 it should be ok.
> 
> That's awesome, happy to hear that!

I don't think anyone is actively working on bumping webpack to v5. Would you be willing to do that? (in a different PR maybe)

**Comment by svrnm** [comment]
> I don't think anyone is actively working on bumping webpack to v5. Would you be willing to do that? (in a different PR maybe)

Let me see what needs to be done for that and if I have the capability and capacity for it.

**Comment by svrnm** [comment]
Ignore my last commit, looks like I did something wrong 🤦‍♂️

**Comment by SimenB** [comment]
Still interesting in seeing this land 😀 anything I can do to help it along? https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1486? it was closed without attention…

**Comment by SimenB** [comment]
Seems that other PR just needed `webpack` in order to work work with `karma`. Seeing as `karma` is [deprecated](https://github.com/karma-runner/karma/pull/3846), it might be better to just migrate to another test runner that doesn't `webpack` to begin with.

**Comment by svrnm** [comment]
> Seems that other PR just needed `webpack` in order to work work with `karma`. Seeing as `karma` is [deprecated](https://github.com/karma-runner/karma/pull/3846), it might be better to just migrate to another test runner that doesn't `webpack` to begin with.

wow, highly appreciated @SimenB !

**Comment by JamieDanielson** [comment]
Sorry on the delay on this. It's been several months since this was last updated, but @svrnm are you still interested in maintaining this? It seems like the blocker right now is to update web tests to use web-test-runner instead of karma. @SimenB already got the updated test runner in #1816 and there is an open PR with an error #2005.

We'll try to take a look at #2005 either way and as a follow-up will want to get the rest of the web components updated to use web-test-runner instead of karma. But still it will be useful to know if you are still interested in maintaining this or if it's no longer a need right now.

**Comment by svrnm** [comment]
> Sorry on the delay on this. It's been several months since this was last updated, but @svrnm are you still interested in maintaining this? It seems like the blocker right now is to update web tests to use web-test-runner instead of karma. @SimenB already got the updated test runner in #1816 and there is an open PR with an error #2005.
> 
> We'll try to take a look at #2005 either way and as a follow-up will want to get the rest of the web components updated to use web-test-runner instead of karma. But still it will be useful to know if you are still interested in maintaining this or if it's no longer a need right now.

I am happy to maintain this, if the tests are unblocked now, or if this requires me to rewrite the tests with web-test-runner instead of karma I can look into that.

**Comment by JamieDanielson** [comment]
> I am happy to maintain this, if the tests are unblocked now, or if this requires me to rewrite the tests with web-test-runner instead of karma I can look into that.

Ok thanks for the confirmation! On #2005 we seem to have found the cause of the errors on updating to use the new test runner, so we should be able to keep moving forward on those. If that's an area you're familiar with it's always useful to have your input or feedback on those changes 😃  

We'll probably want to add a commit or otherwise kick the CI to see the latest on lint and test results.

**Comment by pichlermarc** [comment]
We've hit some roadblocks with updating to `@web/test-runner` (missing named exports from the `shimmer` package, which don't play well with rollup), so I tried again to update webpack to v5 as we've done that in the core repo previously. Workflows are passing, so this might unblock this PR. See #2269

**Comment by svrnm** [comment]
That's great news, @pichlermarc! I'll look into it eventually to fix the extension

**Comment by svrnm** [comment]
FYI, since the code hasn't been touched for that long it will take me some time to get everything back in place

**Comment by svrnm** [comment]
Note, that after discussing this with @JamieDanielson and @pichlermarc I will not bring back this browser extension to reduce the maintenance burden for the maintainers and myself. Instead I work towards listing existing extensions (like https://github.com/tbrockman/opentelemetry-browser-extension) on the [Registry](https://opentelemetry.io/ecosystem/registry/) and then link out to the registry via the README of the archived extension.

**Pull request closed without merging**
